PadMessageHandler Crash

Latest develop

```
Feb 02 11:04:48 etherpad fastRun.sh[36480]: [2021-02-02 11:04:48.461] [ERROR] console - TypeError: Cannot read property 'id' of undefined
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at /home/etherpad-lite/src/node/handler/PadMessageHandler.js:1127:22
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at Array.map (<anonymous>)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at handleClientReady (/home/etherpad-lite/src/node/handler/PadMessageHandler.js:1125:47)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at runMicrotasks (<anonymous>)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at processTicksAndRejections (internal/process/task_queues.js:97:5)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at async Object.exports.handleMessage (/home/etherpad-lite/src/node/handler/PadMessageHandler.js:252:5)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at async Socket.<anonymous> (/home/etherpad-lite/src/node/handler/SocketIORouter.js:76:7)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: [2021-02-02 11:04:48.648] [ERROR] console - TypeError: Cannot read property 'id' of undefined
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at handleClientReady (/home/etherpad-lite/src/node/handler/PadMessageHandler.js:920:39)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at runMicrotasks (<anonymous>)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at runNextTicks (internal/process/task_queues.js:62:5)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at processImmediate (internal/timers.js:429:9)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at process.topLevelDomainCallback (domain.js:137:15)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at async Object.exports.handleMessage (/home/etherpad-lite/src/node/handler/PadMessageHandler.js:252:5)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: at async Socket.<anonymous> (/home/etherpad-lite/src/node/handler/SocketIORouter.js:76:7)
Feb 02 11:04:48 etherpad fastRun.sh[36480]: [2021-02-02 11:04:48.649] [ERROR] console - Error occurred while waiting to exit. Forcing an immediate unclean exit...
Feb 02 11:04:48 etherpad systemd[1]: etherpad.service: main process exited, code=exited, status=1/FAILURE
Feb 02 11:04:48 etherpad systemd[1]: Unit etherpad.service entered failed state.
Feb 02 11:04:48 etherpad systemd[1]: etherpad.service failed.
```

No steps to replicate.

--- a/src/node/handler/PadMessageHandler.js
+++ b/src/node/handler/PadMessageHandler.js
@@ -1438,14 +1438,13 @@ exports.padUsers = async (padID) => {
   await Promise.all(_getRoomSockets(padID).map(async (roomSocket) => {
     const s = sessioninfos[roomSocket.id];
     if (s) {
-      return authorManager.getAuthor(s.author).then((author) => {
-        // Fixes: https://github.com/ether/etherpad-lite/issues/4120
-        // On restart author might not be populated?
-        if (author) {
-          author.id = s.author;
-          padUsers.push(author);
-        }
-      });
+      const author = await authorManager.getAuthor(s.author);
+      // Fixes: https://github.com/ether/etherpad-lite/issues/4120
+      // On restart author might not be populated?
+      if (author) {
+        author.id = s.author;
+        padUsers.push(author);
+      }
     }
   }));
 

--- a/src/node/handler/PadMessageHandler.js
+++ b/src/node/handler/PadMessageHandler.js
@@ -1409,16 +1409,14 @@ const composePadChangesets = async (padId, startNum, endNum) => {
 };
 
 const _getRoomSockets = (padID) => {
-  const roomSockets = [];
-  const room = socketio.sockets.adapter.rooms[padID];
-
-  if (room) {
-    for (const id of Object.keys(room.sockets)) {
-      roomSockets.push(socketio.sockets.sockets[id]);
-    }
-  }
-
-  return roomSockets;
+  const ns = socketio.sockets; // Default namespace.
+  const adapter = ns.adapter;
+  // We could call adapter.clients(), but that method is unnecessarily asynchronous. Replicate what
+  // it does here, but synchronously to avoid a race condition. This code will have to change when
+  // we update to socket.io v3.
+  const room = adapter.rooms[padID];
+  if (!room) return [];
+  return Object.keys(room.sockets).map((id) => ns.connected[id]).filter((s) => s);
 };
 
 /**

